multimodal/src/autogluon/multimodal/data/process_image.py

```diff
@@ -339,49 +366,60 @@ def process_one_sample(
         column_start = 0
         for per_col_name, per_col_image_paths in image_paths.items():
             for img_path in per_col_image_paths[: self.max_img_num_per_col]:
-                with warnings.catch_warnings():
-                    warnings.filterwarnings(
-                        "ignore",
-                        message=(
-                            "Palette images with Transparency expressed in bytes should be converted to RGBA images"
-                        ),
-                    )
-                    is_zero_img = False
-                    try:
-                        img = PIL.Image.open(img_path).convert("RGB")
-                    except Exception as e:
-                        if self.missing_value_strategy.lower() == "zero":
-                            logger.debug(f"Using a zero image due to '{e}'")
-                            img = PIL.Image.new("RGB", (self.size, self.size), color=0)
-                            is_zero_img = True
-                        else:
-                            raise e
-
-                if is_training:
-                    img = self.train_processor(img)
+                if self.prefix == "mmdet_image":
+                    data = dict(img_info=dict(filename=img_path), img_prefix=None)
+                    images.append(self.val_processor(data))
                 else:
-                    img = self.val_processor(img)
-
-                if is_zero_img:
-                    zero_images.append(img)
-                else:
-                    images.append(img)
+                    with warnings.catch_warnings():
+                        warnings.filterwarnings(
+                            "ignore",
+                            message=(
+                                "Palette images with Transparency expressed in bytes should be converted to RGBA images"
+                            ),
+                        )
+                        is_zero_img = False
```

Comment:
Is it better to add the `mmdet_image` check here, i.e., after `is_zero_img = False`?

```
if self.prefix == "mmdet_image":
        img = dict(img_info=dict(filename=img_path), img_prefix=None)
```

Reply:
I feel right now is fine given we only process one sample. We can modify here if necessary in next PR when enable multi-sample evaluation.